File: frontend/lib/core/donation/models/donation.dart
import 'package:equatable/equatable.dart';

enum PaymentMethod { bkash, nagad, rocket, bank, cash }

class Donation extends Equatable {
  const Donation({
    required this.id,
    required this.amount,
    required this.method,
    required this.date,
    required this.status,
  });

  final String id;
  final double amount;
  final PaymentMethod method;
  final DateTime date;
  final String status;

  @override
  List<Object?> get props => [id, amount, method, date, status];
}

File: backend/app/models/donation.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.session import Base

class Donation(Base):
    __tablename__ = "donations"

    id = Column(Integer, primary_key=True, index=True)
    donor_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    amount = Column(Float, nullable=False)
    payment_method = Column(String, nullable=False)  # bkash, nagad, rocket, bank, cash
    payment_status = Column(String, default="pending")  # pending, completed, failed
    transaction_id = Column(String, unique=True, index=True)

    created_at = Column(DateTime(timezone=True), server_default=func.now())

    # Relationships
    donor = relationship("User", back_populates="donations")
